include/capypdf.hpp

```diff
@@ -123,6 +112,52 @@ class PageProperties : public CapyC<CapyPDF_PageProperties> {
     }
 };
 
+class DocumentProperties : public CapyC<CapyPDF_DocumentProperties> {
+public:
+    friend class Generator;
+
+    DocumentProperties() {
+        CapyPDF_DocumentProperties *dp;
+        CAPY_CPP_CHECK(capy_document_properties_new(&dp));
+        _d.reset(dp);
+    }
+
+    void set_title(std::string const &title) {
```

Comment:
These should all be `string_view`s so that people can pass in data from custom string types without having to create a copy of it.